### PR TITLE
[FEMDEM] Minor improvement generating DEM

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
That's it

### CODE REVIEW

File: applications/FemToDemApplication/custom_processes/generate_dem_process.cpp [hunk 1]
```diff
@@ -132,14 +132,24 @@ void GenerateDemProcess::CreateDEMParticle(
     rNode.SetValue(IS_DEM, true);
     rNode.SetValue(RADIUS, Radius);
     rNode.SetValue(DEM_PARTICLE_POINTER, spheric_particle);
+
+    // We transfer kinematic information
+    if (rNode.SolutionStepsDataHas(VELOCITY)) {
+        const array_1d<double, 3> vel = rNode.FastGetSolutionStepValue(VELOCITY);
```
Comment (rubenzorrilla) [inline applications/FemToDemApplication/custom_processes/generate_dem_process.cpp]: I think you don't need a copy here.

File: applications/FemToDemApplication/custom_processes/generate_dem_process.cpp [hunk 2]
```diff
@@ -132,14 +132,24 @@ void GenerateDemProcess::CreateDEMParticle(
...
+        array_1d<double, 3>& r_particle_vel = spheric_particle->GetGeometry()[0].FastGetSolutionStepValue(VELOCITY);
+        noalias(r_particle_vel) = vel;
+    }
+    const array_1d<double, 3> displ = rNode.FastGetSolutionStepValue(DISPLACEMENT);
```
Comment (rubenzorrilla) [inline applications/FemToDemApplication/custom_processes/generate_dem_process.cpp]: Neither here.

File: applications/FemToDemApplication/custom_processes/generate_dem_process.cpp [hunk 3]
```diff
@@ -132,14 +132,24 @@ void GenerateDemProcess::CreateDEMParticle(
     rNode.SetValue(IS_DEM, true);
     rNode.SetValue(RADIUS, Radius);
     rNode.SetValue(DEM_PARTICLE_POINTER, spheric_particle);
+
+    // We transfer kinematic information
+    if (rNode.SolutionStepsDataHas(VELOCITY)) {
```
Comment (rubenzorrilla) [inline applications/FemToDemApplication/custom_processes/generate_dem_process.cpp]: This is way inefficient, if your algorithm allows so, I'd check it once for the first node in the mesh.

Comment (AlejandroCornejo) [reply applications/FemToDemApplication/custom_processes/generate_dem_process.cpp]: that's definetly true

File: applications/FemToDemApplication/custom_processes/generate_dem_process.cpp (hunk 1 above)
Comment (AlejandroCornejo) [reply applications/FemToDemApplication/custom_processes/generate_dem_process.cpp]: Yeah I doubted about that... I want to apply to the particle the same value but afterwards should be free to be different. I suppose there is no problem with that

Comment (rubenzorrilla) [reply applications/FemToDemApplication/custom_processes/generate_dem_process.cpp]: Well, both options work, but the point is that you're allocating something only for the sake of assigning the value below. I'm fully convinced that you can achieve de same with a reference (I don't see your point about fixity).

**Comment by AlejandroCornejo** [comment]
@rubenzorrilla  I've updated the code with your suggestions :)

**Comment by AlejandroCornejo** [comment]
ping @rubenzorrilla  :D

**Review by rubenzorrilla — APPROVED** [review]
Done!

**Pull request merged**